kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/NMDbusConnectorTest.java

```diff
@@ -0,0 +1,619 @@
+
+        thenNoExceptionIsThrown();
+        thenNetworkSettingsDoNotChangeForDevice("eth0");
+        thenNetworkSettingsDoNotChangeForDevice("wlan0");
+    }
+
+    @Test
+    public void applyShouldThrowWithNullMap() throws DBusException {
+        givenBasicMockedDbusConnector();
+        givenMockedDevice("eth0", NMDeviceType.NM_DEVICE_TYPE_ETHERNET, NMDeviceState.NM_DEVICE_STATE_DISCONNECTED,
+                true);
+        givenMockedDevice("wlan0", NMDeviceType.NM_DEVICE_TYPE_WIFI, NMDeviceState.NM_DEVICE_STATE_DISCONNECTED, true);
+        givenMockedDeviceList();
+
+        whenApplyIsCalledWith(null);
+
+        thenNullPointerExceptionIsThrown();
+    }
+
+    @Test
+    public void applyShouldThrowWithEmptyMap() throws DBusException {
+        givenBasicMockedDbusConnector();
+        givenMockedDevice("eth0", NMDeviceType.NM_DEVICE_TYPE_ETHERNET, NMDeviceState.NM_DEVICE_STATE_DISCONNECTED,
+                true);
+        givenMockedDevice("wlan0", NMDeviceType.NM_DEVICE_TYPE_WIFI, NMDeviceState.NM_DEVICE_STATE_DISCONNECTED, true);
+        givenMockedDeviceList();
+
+        whenApplyIsCalledWith(new HashMap<String, Object>());
+
+        thenNoSuchElementExceptionIsThrown();
+    }
+
+    @Test
+    public void applyShouldDoNothingWithEnabledUnsupportedDevices() throws DBusException {
+        givenBasicMockedDbusConnector();
+        givenMockedDevice("unused0", NMDeviceType.NM_DEVICE_TYPE_UNUSED1, NMDeviceState.NM_DEVICE_STATE_DISCONNECTED,
+                true);
+        givenMockedDeviceList();
+
+        givenNetworkConfigMapWith("net.interfaces", "unused0");
+        givenNetworkConfigMapWith("net.interface.unused0.config.dhcpClient4.enabled", true);
+        givenNetworkConfigMapWith("net.interface.unused0.config.ip4.status", "netIPv4StatusEnabledWAN");
+
+        whenApplyIsCalledWith(netConfig);
+
+        thenNoExceptionIsThrown();
+        thenNetworkSettingsDoNotChangeForDevice("unused0");
+    }
+
+    @Test
+    public void applyShouldWorkWithEnabledEthernet() throws DBusException {
+        givenBasicMockedDbusConnector();
+        givenMockedDevice("eth0", NMDeviceType.NM_DEVICE_TYPE_ETHERNET, NMDeviceState.NM_DEVICE_STATE_DISCONNECTED,
+                true);
+        givenMockedDeviceList();
+
+        givenNetworkConfigMapWith("net.interfaces", "eth0");
+        givenNetworkConfigMapWith("net.interface.eth0.config.dhcpClient4.enabled", false);
+        givenNetworkConfigMapWith("net.interface.eth0.config.ip4.status", "netIPv4StatusEnabledWAN");
+        givenNetworkConfigMapWith("net.interface.eth0.config.ip4.address", "192.168.0.12");
+        givenNetworkConfigMapWith("net.interface.eth0.config.ip4.prefix", (short) 25);
+        givenNetworkConfigMapWith("net.interface.eth0.config.ip4.dnsServers", "1.1.1.1");
+
+        whenApplyIsCalledWith(this.netConfig);
+
+        thenNoExceptionIsThrown();
+        thenConnectionUpdateIsCalledFor("eth0");
+        thenNetworkManagerCalledActivateConnection("eth0");
```

Comment:
For consistency I would prefer something like:

```suggestion
        thenActivateConnectionWasCalledFor("eth0");
```